### DISCUSSION TITLE
Azure Monitor: Revamp Variable Editor

### DISCUSSION THREAD

**Discussion by sarahzinger** [grafana/grafana#39970] [Data Sources]
Currently we have 2 issues with our Template Variable Editor for Azure Monitor:
- there is an open bug in which users previously were able to set a default workspace which was used for their template variable queries. We removed this option and now template variable queries are always using whatever the first workspace is https://github.com/grafana/grafana/issues/39444
- it remains confusing how to use the [grafana query functions](https://github.com/grafana/grafana/blob/main/docs/sources/datasources/azuremonitor/template-variables.md#template-variables) we've created for template variables. It's easy to mistype, and hard to discover.
- users can not use ARG to template variable queries

We are proposing that we tackle both of these issues in a series of steps:
1) We create a custom Query Editor for Template Variables in the Azure Monitor datasource that lets users chose 2 query types:
    - Logs, which looks exactly like the regular Query Editor when the Logs service is selected and has a button for the new Resource Picker with a default workspace selected if available. This should solve the first bug (#39444)
    - And an input field for grafana template variable query functions. (this does not solve the second issue, but is a good first iteration towards it)
2) We then iterate on that new custom query editor so that instead of an input field which requires user's to look up documentation to use, we create a series of dropdowns that utilize our prebuilt query functions. 
3) We add in ARG as a query type. It behaves the same as the ARG service option in our regular Query Editor

**Comment by mitulk** [comment]
Thanks @sarahzinger. Looking forward to the PR!

**Comment by joshhunt** [comment]
Sounds like ARG template variable queries will resolve the need in https://github.com/grafana/grafana/issues/18063

**Comment by andresmgot** [comment]
1. is already implemented. I have opened issues for 2. and 3.:

https://github.com/grafana/grafana/issues/50233 
https://github.com/grafana/grafana/issues/50231

Please use those issues if further discussion is needed!
